salutations.php

```diff
@@ -288,13 +312,13 @@ function salutation_process_helper($contact_id, $action) {
  *
  * Helper function for salutation updates after contact updates
  */
-function salutation_update($type, $type_field_id, $contact_id, $salutation_option_selected) {
+function salutation_update($type, $type_field_id, $contact_id, $salutation_type, $greeting_type_id, $salutation_id, $current_calculated_salutation) {
```

Comment:
I changed the function signature to match how the API returns the data more closely.  It's a lot of arguments passed but it also helps readability.